### PR TITLE
feat(DENG-3462): add baseline_clients_last_seen_extended_activity containing new activity fields to support KPI supporting metrics

### PR DESCRIPTION
**Pull request by kik-kik**
# feat(DENG-3462): add baseline_clients_last_seen_extended_activity containing new activity fields to support KPI supporting metrics

This view is a temporary solution until we can make all required changes to make this PR work:
https://github.com/mozilla/bigquery-etl/pull/5434

Which would add those fields to the `baseline_clients_last_seen` view directly.

The following fields are being added to support our KPI metric calculations:

- `app_name` - application name (based on application, isp and in Fenix case on distribution_id)
- `activity_segment` - categorise a client based on their activity "levels"
- `is_dau` - sent ping and considered active on current date
- `is_wau` - sent ping and considered active in the last 7 days
- `is_mau` - sent ping and considered active in the last 28 days
- `is_daily_user` - client a baseline sent ping on current date
- `is_weekly_user` - client a baseline sent ping in the 7 days
- `is_monthly_user` - client a baseline sent ping in the last 28 days
- `is_mobile` - indicates if the entry counts towards mobile DAU
- `is_desktop` - indicates if the entry counts towards desktop DAU
    is_desktop - indicates if the entry counts towards desktop DAU

---

Checklist for reviewer:

- [ ] Commits should reference a bug or github issue, if relevant (if a bug is referenced, the pull request should include the bug number in the title).
- [ ] If the PR comes from a fork, trigger integration CI tests by running the [Push to upstream workflow](https://github.com/mozilla/bigquery-etl/actions/workflows/push-to-upstream.yml) and provide the `<username>:<branch>` of the fork as parameter. The parameter will also show up
in the logs of the `manual-trigger-required-for-fork` CI task together with more detailed instructions.
- [ ] If adding a new field to a query, ensure that the schema and dependent downstream schemas have been updated.
- [ ] When adding a new derived dataset, ensure that data is not available already (fully or partially) and recommend extending an existing dataset in favor of creating new ones. Data can be available in the [bigquery-etl repository](https://github.com/mozilla/bigquery-etl), [looker-hub](https://github.com/mozilla/looker-hub) or in [looker-spoke-default](https://github.com/mozilla/looker-spoke-default/tree/e1315853507fc1ac6e78d252d53dc8df5f5f322b).

For modifications to schemas in restricted namespaces (see [`CODEOWNERS`](https://github.com/mozilla/bigquery-etl/blob/main/CODEOWNERS)):
- [ ] Follow the [change control procedure](https://docs.google.com/document/d/1TTJi4ht7NuzX6BPG_KTr6omaZg70cEpxe9xlpfnHj9k/edit#heading=h.ttegrcfy18ck)

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/DENG-3649)

### CODE REVIEW

File: sql/moz-fx-data-shared-prod/firefox_ios/baseline_clients_last_seen_enriched/view.sql
Comment (bochocki) [inline sql/moz-fx-data-shared-prod/firefox_ios/baseline_clients_last_seen_enriched/view.sql:30]: Something to note: `BIT_COUNT(days_active_bits)` can resolve to 0 if the user is sending pings but does not meet the criteria for qualified usage (i.e. `BIT_COUNT(days_seen_bits) > 0`). In this case those users will end up in the 'other' bucket. I think this is fine, actually, because (1) they are users that don't meet any of the activity criteria and (2) we don't use these activity fields very often. But it's something to keep in mind as we switch from using `days_seen_bits` to `days_active_bits` here -- we're going to see inflation in the "other" category that we wouldn't have seen before the change.

**Review by bochocki — APPROVED** [review]
This looks good to me. Are there more details somewhere about how this will unblock https://github.com/mozilla/bigquery-etl/pull/5434? It wasn't clear to me from that PR what the issue was.

File: sql/moz-fx-data-shared-prod/firefox_ios/baseline_clients_last_seen_enriched/view.sql
Comment (kik-kik) [reply sql/moz-fx-data-shared-prod/firefox_ios/baseline_clients_last_seen_enriched/view.sql:30]: This specific part has been proposed by @bani so I assume this is something she already thought about. The general idea here is that this should be flexible and this specific value will be calculated at runtime.

**Comment by kik-kik** [comment]
> This looks good to me. Are there more details somewhere about how this will unblock #5434? It wasn't clear to me from that PR what the issue was.

@bochocki A small clarification, this will not unblock the other PR. But rather act as a short term solution for us to move forward until the other PR can be merged. Once that is done I will swap out the references in the downstream tables and delete this temporary view. I hope that clears things up.

File: sql/moz-fx-data-shared-prod/firefox_ios/baseline_clients_last_seen_enriched/view.sql
Comment (bani) [reply sql/moz-fx-data-shared-prod/firefox_ios/baseline_clients_last_seen_enriched/view.sql]: I don't remember the conversation about activity segments and it isn't something I have worked with before, but I think classifying as other should work.

Comment (kik-kik) [reply sql/moz-fx-data-shared-prod/firefox_ios/baseline_clients_last_seen_enriched/view.sql]: hmm, ok. Apologies @bani, my memory must have let me down here. I recall us having conversations around keeping this in the views and not the queries, maybe that's why I thought you know more about them.

Comment (bani) [reply sql/moz-fx-data-shared-prod/firefox_ios/baseline_clients_last_seen_enriched/view.sql]: Right, that was for lifecycle stage, which is kind of similar. Activity segment is also probably best left in a view as it's not super set in stone.

**Pull request merged**
